test/screens/play/play_page_test.dart

```diff
@@ -75,4 +76,24 @@ void main() {
     final specialCard = tester.widget<Card>(cardFinder);
     expect(specialCard.color, specialCardColour);
   });
+
+  testWidgets('should display correct count in deck and discard piles', (WidgetTester tester) async {
+    //ugly hack to set a stupid size, so this test doesn't complain about random overflows
+    tester.binding.window.physicalSizeTestValue = Size(2000, 1920);
+
+    final state = makeGameState();
+    state.nextCard();
+    state.nextCard();
+
+    await tester.pumpWidget(createWidgetForTesting(child: PlayPage('New Game', state)));
+
+    final deckFinder = find.widgetWithText(CardPile, 'Deck: ' + state.deck.length.toString());
+    final discardFinder = find.widgetWithText(CardPile, 'Discard: ' + state.discard.length.toString());
+
+    final deckPile = tester.widget<CardPile>(deckFinder);
+    expect(deckPile.pileSize.toString(), state.deck.length.toString());
+
+    final discardPile = tester.widget<CardPile>(discardFinder);
+    expect(discardPile.pileSize.toString(), state.discard.length.toString());
```

Comment:
Kinda duplicating the text checking here - given we've already found them for the exact text (including the number), we could just expect `findsOneWidget` for each of them IMO